compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1647,25 +1647,19 @@ impl<'context> Elaborator<'context> {
             self.push_err(ResolverError::AbiAttributeOutsideContract { span });
         }
 
-        if !let_stmt.comptime && matches!(let_stmt.pattern, Pattern::Mutable(..)) {
+        let explicit_comptime = let_stmt.explicit_comptime;
+        if !explicit_comptime && matches!(let_stmt.pattern, Pattern::Mutable(..)) {
```

Comment:
Instead of erroring here we could error on the parser side. I think with that there would be no need to track `comptime` and `explicit_comptime` separately. Or at least this is the only use of `LetStatement.explicit_comptime` that I found.

Reply:
Or... if we'll consider all globals as `comptime`, do we still need to tag them as such?

Reply:
> Instead of erroring here we could error on the parser side.

I think validating here makes sense:
- Since we don't have a spec of what the elaborator expects to be handled/rejected by the parser
- To make it easier to fuzz/test from a parsed AST in the future

Should I add an additional error in the parser?

> Or at least this is the only use of LetStatement.explicit_comptime that I found.

`explicit_comptime` is primarily used for the formatter, i.e. a global `FOO` with `explicit_comptime=true` will be printed as `comptime FOO..`.

> Or... if we'll consider all globals as comptime, do we still need to tag them as such?

I replaced every occurence of `comptime` that's only used for globals with `explicit_comptime`, but the remaining occurrences are in let statements. I believe the possibilities are:

|                | explicit_comptime=true                 | explicit_comptime=false |
| -------------- | -------------------------------------- | ----------------------- |
| comptime=true  | comptime let foo / comptime global FOO | global FOO              |
| comptime=false | N/A                                    | let foo                 |

Reply:
Thanks!

There's one LSP test that's failing because it's showing a global as `comptime global` instead of `global`. The problem is that the global definition doesn't keep track of whether it's an explicit comptime or not. So I guess we'd need to track that there too.

Reply:
Yes, I ended up added `explicit_comptime` to the HIR let statement definition to fix the hover LSP test. Though r.e. Jake's comment, I'm going to rename it to something like `is_global_let`.